app/utils/SIREN.scala

```diff
@@ -0,0 +1,33 @@
+package utils
+
+import play.api.libs.json._
+import repositories.PostgresProfile.api._
+
+case class SIREN(value: String) {
+  override def toString = value
+}
+
+object SIREN {
+  def apply(value: String) = new SIREN(value.replaceAll("\\s", ""))
+
+  def apply(siret: SIRET) = new SIREN(siret.value.substring(0, 9))
+
+  def pattern = "[0-9]{9}"
```

Comment:
You don't want to do "^[0-9]{9}$" To be sure that a number with more than 9 characters win't match ?
In javascript, at least, `/[0-9]{9}/.test("12345678900000")` returns me true.

Reply:
Not the same in scala
`scala> "12345678900000".matches("[0-9]{9}")`
`res: Boolean = false`